src/javascript.rs

```diff
@@ -111,6 +118,40 @@ fn module_function<'a>(
     ])
 }
 
+fn external_function<'a, T>(
+    public: bool,
+    name: &'a str,
+    arguments: &'a [ExternalFnArg<T>],
+    module: &'a str,
+    fun: &'a str,
+) -> Output<'a> {
+    let head = if public {
+        "export function "
+    } else {
+        "function "
+    };
+    let arguments = wrap_args(arguments.iter().enumerate().map(|a| match a {
```

Comment:
I tried to look for a clever way to reuse some of the code in `module_function` but in the end just keeping it separate seemed clearest.

Reply:
I like this aye